sdk/scala-sdk/src/main/scala/kalix/scalasdk/Kalix.scala

```diff
@@ -157,7 +161,8 @@ class Kalix private (private[kalix] val delegate: javasdk.Kalix) {
    *   a CompletionStage which will be completed when the server has shut down.
    */
   def start(): Future[Done] = {
-    createRunner().run()
+    val conf = ConfigFactory.load()
+    createRunner(sdkNameConfig.withFallback(conf)).run()
```

Comment:
Another option would be to simply remove it from config entirely and only have the sdk names in code. It's not like it is something meant for users to configure/override anyway.

Reply:
yes, that might be easier, and `createRunner()` is public api so this config override might not be used in all cases anyway

Reply:
Right, that definitely makes sense (to avoid the config being exposed and to make sure we have that info regardless of how the user uses the public API). 

In that case, I think we'll need to pass the BuildInfo (or similar) to the javasdk.KalixRunner and then onwards to the DiscoveryImpl. Let me draft that.